File: c/validacao_de_nota.c
#include <stdio.h>

int main(){
    double nota1, nota2, media;

    printf("Digite a primeira nota: ");
    scanf("%lf", &nota1);

    while (nota1 < 0 || nota1 > 10) {
        printf("Valor invalido! Tente novamente: ");
		scanf("%lf", &nota1);
    }

    printf("Digite a segunda nota: ");
    scanf("%lf", &nota2);

    while (nota2 < 0 || nota2 > 10) {
        printf("Valor invalido! Tente novamente: ");
        scanf("%lf", &nota2);
    }

    media = (nota1 + nota2) / 2;

	printf("MEDIA = %.2lf\n", media);

	return 0;
}

File: python/validacao_de_nota.py
nota1: float; nota2: float; media: float

nota1 = float(input("Digite a primeira nota: "))

while nota1 < 0 or nota1 > 10:
	nota1 = float(input("Valor invalido! Tente novamente: "))

nota2 = float(input("Digite a segunda nota: "))

while nota2 < 0 or nota2 > 10:
	nota2 = float(input("Valor invalido! Tente novamente: "))

media = (nota1 + nota2) / 2

print(f"MEDIA = {media:.2f}")